pkg/envoy/server.go

```diff
@@ -238,7 +238,33 @@ func (s *XDSServer) AddListener(name string, endpointPolicyName string, port uin
 
 	listenerConf.FilterChains[0].Filters[1].Config.Fields["http_filters"].GetListValue().Values[0].GetStructValue().Fields["config"].GetStructValue().Fields["policy_name"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: endpointPolicyName}}
 
-	s.listenerMutator.Upsert(ListenerTypeURL, name, listenerConf, []string{"127.0.0.1"}, wg.AddCompletion())
+	retries := 0
+	maxRetries := 10
+	var comp *completion.Completion
+	comp = wg.AddCompletionWithCallbacks(nil, func() {
+		oldPort := port
+		port, err := realloc()
+		if err != nil && retries < maxRetries {
+			retries++
+			listenerConf.Address.GetSocketAddress().PortSpecifier = &envoy_api_v2_core.SocketAddress_PortValue{PortValue: uint32(port)}
+			log.Debugf("Retrying listener %s with reallocated proxyport (%u -> %u), completion %v", name, oldPort, port, comp)
```

Comment:
Is the Completion readable at all in the logs? Is it really useful to log?

Reply:
This was a remnant from debugging, wanted to see it is not `nil`, I'll remove it.